app/controllers/foreman_ansible/api/v2/hostgroups_controller_extensions.rb

```diff
@@ -64,6 +64,30 @@ def assign_ansible_roles
             find_resource
             process_response @hostgroup.update(:ansible_roles => @ansible_roles)
           end
+
+          api :POST, '/hostgroups/:id/add_ansible_roles',
+              N_('Directly add Ansible roles to a hostgroup')
+          param :id, :identifier, :required => true
+          param :ansible_role_ids, Array,
+                N_('Ansible roles to add to a hostgroup'),
+                :required => true
+
+          def add_ansible_roles
+            find_resource
+            process_response @hostgroup.ansible_roles << @ansible_roles
+          end
+
+          api :DELETE, '/hostgroups/:id/remove_ansible_roles',
+              N_('Remove directly assigned Ansible roles from a hostgroup')
+          param :id, :identifier, :required => true
+          param :ansible_role_ids, Array,
```

Comment:
How do you pass an array of ansible_role_ids? I wasn't able to do that (but it works with just one ansible_role_id)

Reply:
You can simply use JSON object with array: { "ansible_role_ids": [1,2,3] }